pkg/controller/storagecluster/storagecluster.go

```diff
@@ -226,6 +246,11 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) waitingForMigrationApproval(cluster *corev1.StorageCluster) bool {
+	approved, err := strconv.ParseBool(cluster.Annotations[constants.AnnotationMigrationApproved])
```

Comment:
so what happens if the annotation doesn't exist? if understand correctly, during operator install it will construct and create a storage cluster with this annotation set to false?

Reply:
Yes, the migration process will create a storagecluster with the annotation set to `false`.

We return `false` only if the annotation is present and set to `false`. That way if this is a migration created cluster, we pause until the user approves it.
If the annotation is absent or has `true` value, we want to go ahead with the normal reconcile loop of creating px pods where absent. I haven't yet figured out how to handle component creation or avoid duplicate components (existing and operator created).